src/util.ts

```diff
@@ -452,7 +462,97 @@ export function createMultiLinkFromLinks(links: string[]): string {
 		hashString += `&${key}=${value}`
 	}
 
-	return url.toString()
+	const shortenedLink = shortenMultilink(url.toString())
+
+	return shortenedLink
+}
+
+export function shortenMultilink(link: string): string {
+	const url = new URL(link)
+	const params = new URLSearchParams(url.search)
+
+	const i = params.get('i')
+	if (i) {
+		const numbers = i
+			.split(',')
+			.map((num) => parseInt(num, 10))
+			.sort((a, b) => a - b)
+
+		let grouped = []
+		let currentGroup = []
+		numbers.forEach((num) => {
+			if (currentGroup.length === 0) {
+				currentGroup.push(num)
+			} else if (num === currentGroup[currentGroup.length - 1] + 1) {
+				currentGroup.push(num)
+			} else {
+				grouped.push(`(${currentGroup[0]},${currentGroup.length})`)
+				currentGroup = [num]
+			}
+		})
+		if (currentGroup.length > 0) {
+			grouped.push(`(${currentGroup[0]},${currentGroup.length})`)
+		}
+
+		let queryString = ''
+		params.forEach((value, key) => {
```

Comment:
Why to make the new url manually instead of using `URLSearchParams` builder? Also I would use `params.set('i', queryString)` instead of iterating through all the parameters